server/shared/src/main/scala/org/http4s/server/staticcontent/FileService.scala

```diff
@@ -70,10 +96,102 @@ object FileService {
         request,
       )
 
+    @deprecated(
+      "Config is no longer a case class. The copy method is provided for binary compatibility.",
+      "0.23.8",
+    )
+    def copy(
+        systemPath: String = this.systemPath,
+        pathCollector: PathCollector[F] = this.pathCollector,
+        pathPrefix: String = this.pathPrefix,
+        bufferSize: Int = this.bufferSize,
+        cacheStrategy: CacheStrategy[F] = this.cacheStrategy,
+    ): Config[F] = new Config[F](
+      systemPath,
+      pathCollector2,
+      pathPrefix,
+      cacheStrategy,
+      bufferSize,
+    )
+
+    @deprecated(
+      "Config is no longer a case class. The productArity method is provided for binary compatibility.",
+      "0.23.8",
+    )
+    override def productArity: Int = 5
+
+    @deprecated(
+      "Config is no longer a case class. The productElement method is provided for binary compatibility.",
+      "0.23.8",
+    )
+    override def productElement(n: Int): Any = n match {
+      case 0 => systemPath
+      case 1 => pathCollector2
+      case 2 => pathPrefix
+      case 3 => cacheStrategy
+      case 4 => bufferSize
+    }
+
+    override def canEqual(that: Any): Boolean = that match {
+      case _: Config[_] => true
+      case _ => false
+    }
+
+    override def equals(other: Any): Boolean = other match {
+      case that: Config[_] =>
+        systemPath == that.systemPath &&
+          pathCollector2 == that.pathCollector2 &&
+          pathPrefix == that.pathPrefix &&
+          cacheStrategy == that.cacheStrategy &&
+          bufferSize == that.bufferSize
+      case _ => false
+    }
+
+    override def hashCode(): Int = {
+      val state = Seq(systemPath, pathCollector2, pathPrefix, cacheStrategy, bufferSize)
+      state.map(_.hashCode()).foldLeft(0)((a, b) => 31 * a + b)
+    }
```

Comment:
Ooph, this was a yak. Looks like you did a fantastic job preserving behavior here, serious props.

Reply:
IDEA helped me with this one :) (adding `[Any]` in the next commit (together with an extra `@nowarn`)  to make scalac happy.